src/rust/metric-forwarder/src/deser_logs_data.rs

```diff
@@ -0,0 +1,58 @@
+use crate::error::MetricForwarderError;
+use crate::error::MetricForwarderError::{
+    DecodeBase64Error, GunzipToStringError, ParseStringToLogsdataError, PoorlyFormattedEventError,
+};
+use aws_lambda_events::event::cloudwatch_logs::{CloudwatchLogsData, CloudwatchLogsEvent};
+
+fn parse_string_to_logsdata(gunzipped: String) -> Result<CloudwatchLogsData, MetricForwarderError> {
+    use serde_json::from_str;
+
+    from_str(&gunzipped).map_err(|err| ParseStringToLogsdataError(err.to_string()))
+}
+
+fn base64_decode_raw_log_to_gzip(data: &str) -> Result<Vec<u8>, MetricForwarderError> {
+    use base64::decode;
```

Comment:
i actually sort of like these scoped imports, makes it immediately obvious why they're pulled in